activerecord/lib/active_record/scoping/named.rb

```diff
@@ -151,7 +170,9 @@ def default_extensions # :nodoc:
         #
         #   Article.published.featured.latest_article
         #   Article.featured.titles
-        def scope(name, body, &block)
+        def scope(name, body, **options, &block)
+          predicate = options.delete(:predicate)
```

Comment:
We can't just a keyword argument here?

```suggestion
        def scope(name, body, predicate: nil, &block)
```

Reply:
I think I was getting ahead of myself, and wanted to offer a flexible interface should more options be added in the future. 

Is it best practice to first start with a keyword argument, and then switch to `**options` if additional arguments are added?